### DISCUSSION TITLE
Single optima instead of non-dominated solutions

### DISCUSSION THREAD

**Discussion by rabiu42000** [anyoptimization/pymoo#222] [Q&A]
What causes  multi or many objectives to return a single optima instead of non-dominated solutions? I am getting a single optima when i use NSGAII or NSGAII for a 3-objective optimization.

**Comment by blankjul** [comment]
If you have no constraints then the most likely reason is that your objectives are simply not conflicting (there exists one solution dominating all others).

**Reply by rabiu42000** [reply]
I have constraints  in  the optimization problem.

**Reply by blankjul** [reply]
if no feasible solution can be found, the least feasible solution might be returned (this option needs to be activated though).

**Reply by rabiu42000** [reply]
I have that option activated. All constraint are not violated. It returns a result but was expecting more than one (non-dominated solutions) because is multi-objective.

**Reply by blankjul** [reply]
then, I my best guess is that either the constraints cause a diversity loss, or your objective are not conflicting.

**Reply by rabiu42000** [reply]
I think that should solve my problem. I will take a look into my objectives again. Thank you very much.
